tendermint/src/rpc/event_listener.rs

```diff
@@ -0,0 +1,77 @@
+//! Tendermint Websocket event listener client
+
+use crate::{
+    net,
+    rpc::Request,
+    rpc::{endpoint::subscribe, Error as RPCError},
+};
+use async_tungstenite::{tokio::connect_async, tokio::TokioAdapter, tungstenite::Message};
+use futures::prelude::*;
+use tokio::net::TcpStream;
+
+/// Event Listener over webocket. https://docs.tendermint.com/master/rpc/#/Websocket/subscribe
+pub struct EventListener {
+    socket: async_tungstenite::WebSocketStream<TokioAdapter<TcpStream>>,
+}
+
+impl EventListener {
+    /// Constructor for event listener
+    pub async fn connect(
```

Comment:
Why not `new`?

Reply:
To me `connect` is fine here, since the method actually initializes a connection by calling `connect_async`.
In general I feel like `new` should be reserved for non-`async` constructors.